service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/digitaltwin/DigitalTwinAsyncClient.java

```diff
@@ -0,0 +1,229 @@
+import com.microsoft.azure.sdk.iot.service.digitaltwin.models.DigitalTwinUpdateRequestOptions;
+import com.microsoft.rest.*;
+import com.microsoft.rest.serializer.JacksonAdapter;
+import lombok.Builder;
+import lombok.NonNull;
+import lombok.Setter;
+import rx.Observable;
+import rx.schedulers.Schedulers;
+
+import java.util.List;
+import java.util.Objects;
+
+import static com.microsoft.azure.sdk.iot.service.digitaltwin.helpers.Tools.*;
+import static lombok.AccessLevel.PACKAGE;
+
+public class DigitalTwinAsyncClient {
+    @Setter(PACKAGE)
+    private DigitalTwinsImpl digitalTwin;
+    private static ObjectMapper objectMapper = new ObjectMapper();
+
+    /***
+     * Creates an implementation instance of {@link DigitalTwins} that is used to invoke the Digital Twin features
+     * @param connectionString The IoTHub connection string
+     */
+    @Builder(builderMethodName = "buildFromConnectionString", builderClassName = "FromConnectionStringBuilder")
+    public DigitalTwinAsyncClient(@NonNull String connectionString) {
+        ServiceConnectionString serviceConnectionString = ServiceConnectionStringParser.parseConnectionString(connectionString);
+        SasTokenProvider sasTokenProvider = serviceConnectionString.createSasTokenProvider();
+        String httpsEndpoint = serviceConnectionString.getHttpsEndpoint();
+
+        init(sasTokenProvider, httpsEndpoint);
+    }
+
+    private void init(SasTokenProvider sasTokenProvider, String httpsEndpoint) {
+        RestClient simpleRestClient = new RestClient.Builder()
+                .withBaseUrl(httpsEndpoint)
+                .withCredentials(new ServiceClientCredentialsProvider(sasTokenProvider))
+                .withResponseBuilderFactory(new ServiceResponseBuilder.Factory())
+                .withSerializerAdapter(new JacksonAdapter())
+                .build();
+
+        IotHubGatewayServiceAPIsImpl protocolLayerClient = new IotHubGatewayServiceAPIsImpl(simpleRestClient);
+        digitalTwin = new DigitalTwinsImpl(simpleRestClient.retrofit(), protocolLayerClient);
+    }
+
+    /**
+     * Gets a digital twin.
+     * @param digitalTwinId The Id of the digital twin.
+     * @param clazz The class to deserialize the application/json into.
+     * @param <T> The generic type to deserialize the application/json into.
+     * @return The application/json of the digital twin.
+     */
+    public <T> Observable<T> getDigitalTwin (@NonNull String digitalTwinId, Class<T> clazz)
+    {
+        return digitalTwin.getDigitalTwinAsync(digitalTwinId)
+                .filter(Objects::nonNull)
+                .map(response -> {
+                    T genericResponse = null;
+                    try {
+                        genericResponse = DeserializationHelpers.castObject(objectMapper, response, clazz);
+                    } catch (JsonProcessingException e) {
+                        Observable.error(e);
```

Comment:
if an exception is thrown and caught here, it seems like we are returning null (through 'return generticResponse'). Is this intentional? or do we expect to rethrow the exception to the user?